app/services/idv/steps/doc_auth_base_step.rb

```diff
@@ -95,7 +95,7 @@ def throttled_url
 
       def throttled_else_increment
         Throttle.for(
-          target: user_id,
+          user: current_user,
```

Comment:
`current_user` doesn't always exist. See `#user_id`

Reply:
ah right, have to re-implement `EffectiveUser`

Reply:
PTAL c875a2a40abb38f53f73325f6a9915b42726efac

Reply:
Can we just use `EffectiveUser` ? Currently trying to sift through difference between `flow_session` and `session`.

c875a2a could work, but it'd be nice if we can avoid a DB call for `User.find` for current user like we do in `EffectiveUser`.

Reply:
https://github.com/18F/identity-idp/blob/a0fb1892b7db8695725f237bf415e36ff8a14778/app/services/flow/base_flow.rb#L13
comes from
https://github.com/18F/identity-idp/blob/a0fb1892b7db8695725f237bf415e36ff8a14778/app/services/flow/flow_state_machine.rb#L86
which comes from
https://github.com/18F/identity-idp/blob/a0fb1892b7db8695725f237bf415e36ff8a14778/app/services/flow/flow_state_machine.rb#L223-L225

So since `user_session` is sometimes nested, looks like no we can't just blindly use `EffectiveUser` :[

Reply:
New change: 1763f5e099a54451b072140864cf9a647db6d55f

Reply:
> So since `user_session` is sometimes nested, looks like no we can't just blindly use `EffectiveUser` :[

Ah, okay. That makes more sense. Is it reasonable that EffectiveUser could also try to pick from `user_session`?

Reply:
If we really wanted to re-use the EffectiveUser mixin I bet we could modify it like this?

```diff
   def effective_user_id
     [
+      respond_to?(:flow_session) && flow_session[:doc_capture_user_id],
       session[:doc_capture_user_id],
       current_user&.id,
     ].find(&:present?)
   end
```

Reply:
> If we really wanted to re-use the EffectiveUser mixin I bet we could modify it like this?

Yeah, I think the idea with the mixin is to try a few potential places to grab the "current" user. Your suggestion is a bit more involved than I'd expected (vs. `user_session[:doc_capture_user_id]` I'd hoped it was). I'd be fine to just leave things as-is.

Reply:
one issue is that the steps/flow/world is very controller-like but are not actually controllers :[